src/printer/smt2/smt2_printer.cpp

```diff
@@ -206,6 +206,19 @@ void Smt2Printer::toStream(std::ostream& out, Kind k) const
   out << smtKindString(k);
 }
 
+void toStreamUninterpretedSortValue(std::ostream& out,
+                                    const UninterpretedSortValue& v)
+{
+  if (v.getType().isInstantiated())
+  {
+    out << "|" << v << "|";
```

Comment:
There is a standard place for doing this, https://github.com/cvc5/cvc5/blob/main/src/util/smt2_quote_string.cpp#L26.
You should generate the name and then run it through that utility which will determine if it needs to be quoted.